pkg/systemd/reporting.jsx

```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
                             return;
                         }
 
-                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", this.props.workflow[0], (ex.problem || ex.message)));
-                        this.setState({ message: _("Reporting failed") });
+                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", workflow[0], (ex.problem || ex.message)));
+                        setMessage(_("Reporting failed"));
                     });
                 })
                 .catch(ex => console.error(cockpit.format("Setting up a D-Bus proxy for $0 failed: $1", object_path, ex)));
-    }
-
-    _onReportButtonClick(event) {
-        this.setState({ problemState: ProblemState.UNREPORTABLE });
+    };
 
-        this.setState({
-            message: _("Waiting to start…"),
-            problemState: ProblemState.REPORTING,
-        });
+    const onReportButtonClick = (_event) => {
+        setProblemState(ProblemState.UNREPORTABLE);
+        setMessage(_("Waiting to start…"));
+        setProblemState(ProblemState.REPORTING);
 
-        this.props.client
-                .wait()
+        client.wait()
                 .catch(exception => console.error(cockpit.format("Channel for reportd D-Bus client closed: $0", exception.problem || exception.message)))
-                .then(() => this._createTask(this.props.client))
+                .then(() => createTask(client))
                 .catch(exception => {
-                    const message = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
-
-                    this.setState({
-                        message,
-                        problemState: ProblemState.REPORTABLE,
-                    });
+                    setMessage(cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message)));
+                    setProblemState(ProblemState.REPORTABLE);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)